src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -32,25 +32,74 @@ public void createNewUserProgress(final String chatID) throws UtilsborException
 //        userProgressStorage.save(userProgress);
     }
 
-    public String processTransportType(final Command requestingCommand, final String chatID) throws UtilsborException {
+    public String processGeneralTransportType(final Command requestingCommand, final String chatID) throws UtilsborException {
         final UserProgress userProgress;
         userProgress = userProgressStorage.get(chatID);
-        final String message;
 
-        if (Objects.equals(requestingCommand, Command.M1)) {
-            userProgress.setTransportType(TransportType.M1);
+        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
+            throw new InvalidOrderCommandException(chatID, requestingCommand);
+        }
+
+        userProgress.setNextStep(requestingCommand.getNextStep());
+
+        switch (requestingCommand) {
+            case M1 -> userProgress.setGeneralTransportType(GeneralTransportType.M1);
+            case BUSES_AND_TRUCKS -> userProgress.setGeneralTransportType(GeneralTransportType.EXCEPT_M1);
         }
 
         userProgressStorage.save(userProgress);
-        message = messagesCreator.getSummaryAnswer(userProgress);
-        return message;
+        return messagesCreator.getSummaryAnswer(userProgress);
+    }
+
+    public String processN1_N3TransportType(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress;
+        userProgress = userProgressStorage.get(chatID);
+
+        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
+            throw new InvalidOrderCommandException(chatID, requestingCommand);
+        }
+        userProgress.setNextStep(requestingCommand.getNextStep());
+        switch (requestingCommand) {
+            case N1_N3 -> userProgress.setExceptM1TransportType(ExceptM1TransportType.N1_N3);
+            case M2_M3 -> userProgress.setExceptM1TransportType(ExceptM1TransportType.M2_M3);
+            case TRUCK_UNITS -> userProgress.setExceptM1TransportType(ExceptM1TransportType.TRUCK_UNITS);
+            case TRAILERS_O4 -> userProgress.setExceptM1TransportType(ExceptM1TransportType.TRAILERS_O4);
+        }
+
+        userProgressStorage.save(userProgress);
+        return messagesCreator.getSummaryAnswer(userProgress);
+    }
+
+    public String processN1_N3TransportWeight(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+
+        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
+            throw new InvalidOrderCommandException(chatID, requestingCommand);
+        }
+        userProgress.setNextStep(requestingCommand.getNextStep());
+        switch (requestingCommand) {
+            case LESS_2_TONS -> userProgress.setTransportWeightN1N2N3(N1N3TransportWeight.LESS_2_TONS);
```

Comment:
I like your idea very much. Unfortunately, it seems I have a problem.
We have userProgress object with fields for price calculation. We need to fill fields with 100% accuracy. We need to know which type of vehicle in particular to calculate a correct price. 
But during filling fillFieldsToCommandsMap sometimes we use families instead of particular field. 
For example. For a price calculation I have to know the volume of an engine. But for the fillFieldsToCommandsMap is enough to know only that volume field was filled - no difference, which in particular. 
I see no solutions right now, but we can discuss it and find the best way.